gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -386,16 +400,18 @@ private String startResumableUpload() throws InterruptedException, IOException {
 
       SimpleResponseObserver<StartResumableWriteResponse> responseObserver =
           new SimpleResponseObserver<>();
-
-      stub.withDeadlineAfter(START_RESUMABLE_WRITE_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS)
-          .startResumableWrite(request, responseObserver);
-      responseObserver.done.await();
-
-      if (responseObserver.hasError()) {
-        throw new IOException(
-            String.format("Failed to start resumable upload for '%s'", resourceId),
-            responseObserver.getError());
-      }
+      runWithRetries(
+          () -> {
+            stub.withDeadlineAfter(START_RESUMABLE_WRITE_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS)
+                .startResumableWrite(request, responseObserver);
+            try {
+              responseObserver.done.await();
+            } catch (InterruptedException e) {
+              // Do not throw here since it will eventually throw if all retries fail.
+              e.printStackTrace();
```

Comment:
`InterruptedException` is different than `IOException` and usually this means that such exception should not be swallowed, you need to handle it like this: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/00e3f553ff49a189518206b4d531f6a9906b4d09/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java#L942:L945